articles/151_node_to_participant_mapping.md

```diff
@@ -0,0 +1,241 @@
+From a security point of view, only being able to configure it at a `Participant` (or per process) level is enough.
+There's not much sense on having different access control policies for `Nodes` in the same process.
+As they share the same address space, other vulnerabilities are possible.
+
+##### How to create a new security key?
+
+Before, we were creating a key for each `Node`.
+The full name of the node was used for creating it.
+
+If we create one `Participant` per context, we will only need a key for each of them, and not one per `Node`.
+There are two alternatives:
+
+- Add the concept of `Context` name (or `Participant` name).
+  In this way, the key of each participant could be specified independently.
+- Use one key per process.
+  All the `Participants` within one process will use the same key.
+
+##### How to specify access policies?
+
+Access control policies could still be specified per `Node` basis.
+When a `Participant` is created, it should look at the access control policies of each of its `Nodes` and compose them in a single configuration.
+
+#### Node Name Uniqueness
+
+In `Dashing` and before, `Node` name uniqueness is not enfornced.
+
+When creating only one `Participant` per `Context`, we can distinguish two cases:
+- There is an overlap between the name of two `Nodes` created within the same `Context`.
+  This case can be trivially solved.
+- There is a collision with the `Node` name created from another `Context`.
+  By the nature of discovery, when a collision is detected, it's not possible to know what `Node` was created first without extra information.
+  A collision resolution mechanism have to be decided for solving which `Node` continues living.
+  A `timestamp` of the `Node` creation published in the state message can help to solve the problem.
+
+If we don't change the `Node` to `Participant` mapping, the last item still stands and should be solved in a similar fashion.
+
+#### Ignore local publications option
+
+There's an `ignore_local_publications` option that can be set when [creating a subscription](https://github.com/ros2/rmw/blob/2250b3eee645d90f9e9d6c96d71ce3aada9944f3/rmw/include/rmw/rmw.h#L517).
+That option avoids receiving messages from `Publishers` within the same `Node`.
+This wasn't implemented in all the rmw implementations (e.g.: [FastRTPS](https://github.com/ros2/rmw_fastrtps/blob/099f9eed9a0f581447405fbd877c6d3b15f1f26e/rmw_fastrtps_cpp/src/rmw_subscription.cpp#L118)).
+
+For emulating this behavior, messages could be ignored by checking from what `Node` the `Publisher` was created.
+This should be possible by querying the state messages, or the local cache were they are organized.
+
+
+#### Intra process communication
+
+Currently, intra-process communication can be enable disabled in each `Publisher` and `Subscription`.
+There is only one reason for that: intraprocess communication doesn't support all QoS policies.
```

Comment:
This is an important reason, but I don't know if it is the only reason... Perhaps just say "an important reason for being able to selectively enable intra-process is..."